hazelcast/src/main/java/com/hazelcast/client/config/impl/ClientDomConfigProcessor.java

```diff
@@ -391,9 +393,24 @@ private void handleLoadBalancer(Node node) {
             clientConfig.setLoadBalancer(new RandomLB());
         } else if ("round-robin".equals(type)) {
             clientConfig.setLoadBalancer(new RoundRobinLB());
+        } else if ("custom".equals(type)) {
+            String loadBalancerClassName = parseCustomLoadBalancerClassName(node);
+
+            try {
+                LoadBalancer loadBalancer = newInstance(null, loadBalancerClassName);
```

Comment:
Will this (passing `null` as the class loader) cause problems when a different class loader is provided for the client config? I see that for other elements, we store the class name and later, use the class loader defined in the config (if any) while instantiating the objects.

Reply:
I did not consider the class-loader problem when we are discussing here https://github.com/hazelcast/hazelcast/issues/17228#issuecomment-662378919

@the-thing Lets move the instantiation to `com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl#initLoadBalancer`
To achieve this, we can add a new field to the ClientConfig as loadBalancerClassName. ConfigProcessor sets this class name of the load balancer type is `custom`. 
This way, we will not break the old behavior.

Reply:
Followed the suggested change and move the instantiation of load balancer to `com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl#initLoadBalancer`. I had to rewrite the test cases and created a few new ones in `com.hazelcast.client.config.ClientConfigLoadBalancerTest`. This required some changes to I would advise to review the whole change again.